app/src/main/java/com/hadiyarajesh/notex/repository/NotesRepository.kt

```diff
@@ -1,8 +1,20 @@
 package com.hadiyarajesh.notex.repository
 
+import com.hadiyarajesh.notex.database.dao.NoteDao
+import com.hadiyarajesh.notex.database.entity.Note
+import kotlinx.coroutines.flow.Flow
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
-class NotesRepository @Inject constructor() {
+class NotesRepository @Inject constructor(private val noteDao: NoteDao) {
+
```

Comment:
What about deleting (archiving) note method? 
Also, if note is in any folder and when user update the note, we should automatically update the `updatedOn` property of that folder as well. We need to think about side cases as well. (discussed in call, maybe you were not present, so stating it here.)